app/models/course/group.rb

```diff
@@ -0,0 +1,37 @@
+class Course::Group < ActiveRecord::Base
+  stampable
+  belongs_to :course, inverse_of: :groups
+  has_many :group_users, inverse_of: :group, dependent: :destroy,
+                         class_name: Course::GroupUser.name, foreign_key: :course_group_id
+
+  # Add the given user to this group.
+  #
+  # @param [User] The user to be added. The user must have already been enrolled to the course.
+  # @param [Symbol] The role in group, default is :normal
+  # @return [Boolean] true if added successfully, otherwise false
+  def add_user(user, role = :normal)
+    group_users.build(user: user, role: role).save
```

Comment:
How about <code> role </code> ? I think this function is mainly for assign a role